ratelimit/backends/base.py

```diff
@@ -75,14 +75,6 @@ async def set_block_time(self, user: str, block_time: int) -> None:
     async def is_blocking(self, user: str) -> bool:
         raise NotImplementedError()
 
+    @abstractmethod
     async def allow_request(self, path: str, user: str, rule: Rule) -> bool:
-        if await self.is_blocking(user):
-            return False
-
-        updated = await self.increase_limit(path, user, rule)
-        allow = updated or await self.decrease_limit(path, user, rule)
-
-        if not allow and rule.block_time:
-            await self.set_block_time(user, rule.block_time)
-
-        return allow
+        raise NotImplementedError()
```

Comment:
It seems that there is no need to modify it, this method can be overridden in the subclass.